services/payments/statemachine_zebpay.go

```diff
@@ -62,7 +65,16 @@ func (zm *ZebpayMachine) Pay(ctx context.Context) (*paymentLib.AuthenticatedPaym
 		case zebpay.TransferPendingCode:
 			// Set backoff without changing status
 			zm.backoffFactor = zm.backoffFactor * 2
+		case zebpay.TransferNotFoundCode:
+			// The transaction was never received by zebpay. We should not have gotten into this
+			// condition unless zebpay accepted this transaction, but failed to persist the record.
+			// We'll need to retry the submission and come back around to check status, so do not
+			// SetNextState except those that get set as part of the submit function
+			return zm.transaction, zm.submit(ctx, entry, to)
 		default:
+			if cterr != nil {
+				return zm.transaction, fmt.Errorf("failed to check transaction status: %w", err)
+			}
```

Comment:
Why not check it earlier? Like on line 56? It should not get to the `switch` if there was an error, should it?

Reply:
The `NotFound` result is treated as an error by the HTTP library and the `Client` interface requires us to return an error from `CheckStatus` even though the `ctr` has a field for errors. I imagine changing the `NotFound` result into a non-error would be unexpected. The options available as a result are to have a complex conditional in an `err !=nil` on 56 that ignores the error if it's `NotFound` and then proceeds to the `switch`, or process the valid-to-us `NotFound` status alongside the other `case`s as I've done and handle the error in a non-standard location.